### DISCUSSION TITLE
local supabase won't start `Error status 502: An invalid response was received from the upstream server`

### DISCUSSION THREAD

**Discussion by devidw** [supabase/supabase#29702] [Questions] [answered]
I run into this issue with the latest version, and thought it had something to do with the specific cli version, but after testing beta and older versions and still hitting this I doubt its because of that.

Original issue comment: https://github.com/supabase/cli/issues/2714#issuecomment-2395403059

---

> Error status 502: An invalid response was received from the upstream server

```console
$ sw_vers; docker --version; supabase --version
ProductName:            macOS
ProductVersion:         14.6.1
BuildVersion:           23G93
Docker version 27.1.2, build d01f264bcc
1.200.3
```

```console
$ supabase start
Seeding globals from roles.sql...
Seeding data from seed.sql...
Stopping containers...
Error status 502: An invalid response was received from the upstream server

Try rerunning the command with --debug to troubleshoot the error.
```

Running with `--debug` results in the same error message and does not seem to include more info on the actual issue.

---

Steps I tried without success:

- I am running docker engine on macOS with colima, I completely wiped the entire vm `colima delete -f` and re-created it, this does drop all images and whatnot
- Reinstalled supabase cli `brew uninstall supabase && brew install supabase`
- Created a fresh `supabase init`

None of those fixed the issue.

- Update: Also tried all the above steps with the beta `1.202.4`, but same outcome.
- Update: Hm, even when downgrading to a random older version `1.191.3` I still get the same start-up issue. Wonder what that could cause, as the error message is not very helpful to me

---

After continuing to invest, I can start when the api service is disabled. When I enable it, it fails.

While keeping an eye on the docker containers, it looks like the health check on the rest container is failing, seeing a bunch of connection refused inside the kong container trying to reach the rest container.

However, logs from the rest container look fine as far as I can tell:

```console
$ supabase stop && supabase start
Stopped supabase local development setup.
Local data are backed up to docker volume. Use docker to show them: docker volume ls --filter label=com.supabase.cli.project=sb-test
supabase_rest_sb-test container logs:
08/Oct/2024:09:54:56 +0000: Starting PostgREST 12.2.0 (ec89f6b)...
08/Oct/2024:09:54:56 +0000: Attempting to connect to the database...
08/Oct/2024:09:54:56 +0000: Admin server listening on port 3001
08/Oct/2024:09:54:56 +0000: Listening on port 3000
08/Oct/2024:09:54:56 +0000: Successfully connected to PostgreSQL 15.1 (Ubuntu 15.1-1.pgdg20.04+1) on aarch64-unknown-linux-gnu, compiled by gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0, 64-bit
08/Oct/2024:09:54:56 +0000: Listening for notifications on the "pgrst" channel
08/Oct/2024:09:54:56 +0000: Config reloaded
08/Oct/2024:09:54:56 +0000: Schema cache queried in 2.7 milliseconds
08/Oct/2024:09:54:56 +0000: Schema cache loaded 0 Relations, 0 Relationships, 1 Functions, 0 Domain Representations, 4 Media Type Handlers
Stopping containers...
Error status 502:
Try rerunning the command with --debug to troubleshoot the error.
```

**Comment by devidw** [comment, marked as answer]
Okay turned out that I was connected to some shared network and that somehow messed up the docker network and even restarting docker after switching to a private network did still run with the ips of the shared network. Removing the shared network and restarting the OS was the only thing that resolved it.

**Reply by NicoCaldo** [reply]
I have the same issue. What do you mean by "I was connected to some shared network"? Your OS was connected to a shared network? Like network manager of Windows/Linux?

**Reply by devidw** [reply]
Ah, a shared/public wifi network I meant.
